### ISSUE TITLE
Incorrect Boolean evaluation

### ISSUE THREAD

**Issue by ccaapton** [pyparsing/plusminus#23]
The result of below code evaluation should be true, but it returns False instead.
```
from plusminus import BaseArithmeticParser

parser = BaseArithmeticParser()
parser.evaluate('False or False or True')
```

**Comment by ptmcg** [comment]
Sure enough! I had tried to implement short-circuiting logic. Better to use Python's `any` and `all` builtins instead.

**Comment by ptmcg** [comment]
I have a working fix - are you doing more testing, or do you need it immediately and I'll push out 0.8.1?

**Comment by ccaapton** [comment]
> I have a working fix - are you doing more testing, or do you need it immediately and I'll push out 0.8.1?

I wrote something like below, works for this test case, but not sure if good for all. Can you give me the fix as well?
```
class MyBinaryLogicalOr(BinaryLogicalOperator):
    opns_map = {
        "||": operator.or_
    }
    def left_associative_evaluate(self, oper_fn_map):
        with _trimming_exception_traceback():
            last = bool(self.tokens[0].evaluate())
            ret = False
            for oper, operand in zip(self.tokens[1::2], self.tokens[2::2]):
                next_ = bool(operand.evaluate())
                ret = ret or oper_fn_map[oper](last, next_)
                if ret:
                    return ret
                last = next_
            return ret
```

**Comment by ptmcg** [comment]
I just uploaded the fixed version to pypi, version 0.8.1

**Comment by ptmcg** [comment]
Fixed in version 0.8.1
